nominal/core.py

```diff
@@ -311,6 +328,188 @@ def _from_conjure(cls, client: NominalClient, dataset: scout_catalog.EnrichedDat
+            self.checks, 
+            self._client._auth_header, 
+            self._client._compute_representation_client, 
+            self.default_ref_name
+        )
+    
+        request = scout_checks_api.CreateChecklistRequest(
+            commit_message=commit_message if commit_message is not None else "",
+            assignee_rid=self._client._get_user_rid_from_email(self.assignee_email),
+            title=self.name,
+            description=self.description if self.description is not None else "",
+            # TODO(ritwikdixit): support functions
+            functions={},
+            properties={} if self.properties is None else dict(self.properties),
+            labels=[] if self.labels is None else list(self.labels),
+            checks=conjure_checks,
+            checklist_variables=conjure_checklist_variables,
+            #TODO(ritwikdixit): support checklist VCS
+            is_published=True,
+        )
+
+        response = self._client._checklist_api_client.create(self._client._auth_header, request)
+        return Checklist._from_conjure(self._client, response)
+
+
+# TODO(ritwikdixit): add support for more Checklist metadata and versioning
+@dataclass(frozen=True)
+class Checklist:
+    rid: str
+    name: str
+    description: str
+    properties: Mapping[str, str]
+    labels: Sequence[str]
+    checklist_variables: Sequence[ChecklistVariable]
+    checks: Sequence[Check]
+    _client: NominalClient = field(repr=False)
+
+    @staticmethod
+    def _from_conjure(client: NominalClient, checklist: scout_checks_api.VersionedChecklist) -> Checklist:
+
+        # TODO(ritwikdixit): support draft checklists with VCS
+        if not checklist.metadata.is_published:
+            raise ValueError("cannot get a checklist that has not been published")
+        
+        variable_name_to_graph_map = { variable_name: compute_graph for variable_name, compute_graph in (_conjure_checklist_variable_to_name_graph__pair(checklistVariable)
+            for checklistVariable in checklist.checklist_variables)
+        }
+        check_rid_to_graph_and_def_map = { check_definition.rid: (check_definition, compute_graph) for check_definition, compute_graph in (_conjure_check_to_check_definition_graph_pair(check)
+            for check in checklist.checks)
+        }
+
+        # # TODO(ritwikdixit): remove the need for these extraneous network requests
+        variable_names_to_expressions = client._compute_representation_client.batch_compute_to_expression(
+            client._auth_header,
+            variable_name_to_graph_map
+        )
+        check_rids_to_expressions = client._compute_representation_client.batch_compute_to_expression(
+            client._auth_header,
+            {check_rid : graph for check_rid, (_, graph) in check_rid_to_graph_and_def_map.items()}
+        )
+        check_rids_to_definitions = {check_rid: check_def for check_rid, (check_def, _) in check_rid_to_graph_and_def_map.items()}
```

Comment:
I don't love the style here, but the goals were:

1. Batch the requests for converting between compute graph to expression in such a way that I can always keep track of the graph, expression, and metadata for a rid

2. Remove the concept of the graph from Check and Variable signature for the user

Reply:
Let me know if I should change the dict values being tuples as well